src/main/kotlin/com/wgslfuzz/core/Resolver.kt

```diff
@@ -510,71 +571,30 @@ private fun nodeIntroducesNewScope(
             parentNode !is ContinuingStatement
     )
 
+private fun Type.asStoreType(): Type =
```

Comment:
Nitpicking the naming here: am I right that this doesn't always return a storable type when applied to a type? E.g., when applied to AbstractInt it will return AbstractInt; when applied to a pointer type it returns a pointer type.

Although more verbose, arguably `asStoreTypeIfReference` would be a more accurate name. Or `dereference`, or `removeReference`.

But up to you - fine to keep it as is if you prefer the current name.